lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -434,7 +434,7 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
         // xmrPathSuffix and end the path here.
         auto xmrIter = xmrPathSuffix.find(continueFrom.value());
         if (xmrIter != xmrPathSuffix.end())
-          xmrPathSuffix[indx] = xmrIter->getSecond();
+          xmrPathSuffix[indx] = xmrIter->getSecond().str();
```

Comment:
What does this change / fix?

Reply:
This does fix a GrandCentralDataTaps internalPaths issue, But there is a Valgrind memory error at this location.
I wrongly assumed this would fix it, but after running for a while, the invalid read error still exists. I am trying to figure out what's going on.

Reply:
@dtzSiFive let me know if anything related to `DenseMap` and `SmallString` here looks funky. There is a bug that occurs only for debug builds and not for release build. Valgrind does show this SmallString assignment as an invalid memory read.

Reply:
Do a `try_emplace(indx)` first to add it to the map (maybe before the `find`?), otherwise the DenseMap can/will resize while evaluating the assignment and the iterator will be invalid/problematic.  Not sure there's a good way to copy a DenseMap value (as in key-value "value") to a new key with copy constructor (that is, without constructing it in-place first), but there might be.

Reply:
I guess before the `find` has problem of adding it if the `continueFrom` isn't in the map, not sure if that's what you want, sorry.  But definitely think it's being resized during the assignment.

Reply:
Thanks @dtzSiFive for the suggestion. The reallocation was the issue. I did the naive solution of copying it to a temporary.
This fixes the invalid memory read, the valgrind is now clean.